### PR TITLE
add tutorial text to gui + minor fixes

### PR DESCRIPTION
**Pull request by Tymoteuszauke**

### CODE REVIEW

File: client/src/main/java/com/blackship/battlesheep/communication/RandomizedBoardReceiver.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private final AppClientCommunicationHandler appClientCommunicationHandler;
+
+    public RandomizedBoardReceiver(AppClientCommunicationHandler appClientCommunicationHandler) {
+        this.appClientCommunicationHandler = appClientCommunicationHandler;
+    }
+
+    public List<Integer> receiveRandomizedPlayerBoard() throws IOException, ClassNotFoundException {
+
+        Packet packet = appClientCommunicationHandler.read();
+        return ((PacketBoard)packet).getPlayerPositions()
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/communication/RandomizedBoardReceiver.java:24]: Nice :+1:

File: client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java [hunk 2]
```diff
@@ -18,35 +20,30 @@
...
 
         enemyDestroyedMasts.clear();
         playerDestroyedMasts.clear();
-        PacketMove packet = (PacketMove) appClientCommunicationHandler.read();
+        Packet packet = appClientCommunicationHandler.read();
 
-        if (((Packet)packet).getPacketType() == PacketType.MOVE) {
-            enemyDestroyedMasts.addAll(packet.getPositions().get(1));
-            playerDestroyedMasts.addAll(packet.getPositions().get(0));
+        if (packet.getPacketType() == PacketType.MOVE) handlePacketMove((PacketMove) packet);
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java:42]: I prefer way with curly braces for me it is more readable.

File: client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java [hunk 3]
```diff
@@ -72,4 +69,22 @@ public boolean fireSalvoCannonade(List<Integer> positionsToBeBalled) {
...
+    public boolean hasWinner() {
+        return hasWinner;
+    }
+
+    public String getWinner() {
+        return winner;
+    }
+
+    private void handlePacketMove(PacketMove packetMove) {
+        enemyDestroyedMasts.addAll(packetMove.getPositions().get(1));
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java:82]: Magic numbers.

File: client/src/main/java/com/blackship/battlesheep/fx/board_view_updater/BoardViewUpdater.java [hunk 4]
```diff
@@ -54,23 +55,27 @@ public BoardViewUpdater build() {
...
+                    });
+        }
     }
 
     public void feedCannon(int cannonBall) {
         loadedCannons.add(cannonBall);
         long availableCannons = playerMastPositions
                 .stream()
-                .filter(data -> !data.isDisabled())
+                .filter(data -> !data.getText().equals("X"))
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/board_view_updater/BoardViewUpdater.java:82]: Magic string.

**Review by mhbratek — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java (hunk 3 above)
Comment (Dzionek95) [reply client/src/main/java/com/blackship/battlesheep/fx/SalvoHandler.java:82]: +1

File: client/src/main/java/com/blackship/battlesheep/fx/board_view_updater/BoardViewUpdater.java (hunk 4 above)
Comment (Dzionek95) [reply client/src/main/java/com/blackship/battlesheep/fx/board_view_updater/BoardViewUpdater.java:82]: +1

**Review by Dzionek95 — APPROVED** [review]

**Pull request merged**